runtime/doc/lua.txt

```diff
@@ -2366,10 +2366,29 @@ vim.trim({s})                                                     *vim.trim()*
       • https://www.lua.org/pil/20.2.html
 
 vim.validate({opt})                                           *vim.validate()*
-    Validates a parameter specification (types and values). Specs are
-    evaluated in alphanumeric order, until the first failure.
+    Validate function arguments.
 
-    Usage example: >lua
+    This function has two valid forms:
+    1. vim.validate(name: str, value: any, type: string, optional: bool)
+    2. vim.validate(spec: table)
+
+    Form 1 validates that argument {name} with value {value} has the type
+    {type}. If {optional} is true, then {value} may be null. This form should
+    be preferred for simple cases.
```

Comment:
Would we want to mention that it won't work with "callable" because this path does not use `vim.is_callable`?

Reply:
Mentioned in https://github.com/neovim/neovim/pull/28977/commits/d5d7079557607d124b1d51d90b2e851c1c35104c

Reply:
(and slightly reworded in https://github.com/neovim/neovim/pull/28977/commits/717d170ae0dc753cbb5fac18be0e535c7489d5ae)